src/tpm.rs

```diff
@@ -23,3 +42,53 @@ pub(crate) fn get_tpm2_ctx() -> Result<tss_esapi::Context, tss_esapi::Error> {
     let tcti = Tcti::from_str(tcti_path)?;
     unsafe { Context::new(tcti) }
 }
+
+/*
+ * Input: None
+ * Return:
+ *   TPM Vendor Type
+ *   tss_esapi::Error
+ *
+ * get_tpm_vendor will retrieve the type of TPM. This allows us
+ * to perform opertions, such as understand the host is using a
+ * software TPM as opposed to hardware TPM.
+ */
+pub fn get_tpm_vendor() -> Result<String, tss_esapi::Error> {
+    let mut ctx = get_tpm2_ctx()?;
```

Comment:
I think you want to get the context in `main()` and pass that one through to here as `&mut Context`.